src/useAnswersState.tsx

```diff
@@ -1,15 +1,75 @@
-import { useContext, useRef, useState } from 'react';
+import React, { useCallback, useContext, useEffect, useRef, useState } from 'react';
 import { State } from '@yext/answers-headless';
 import { AnswersHeadlessContext } from './AnswersHeadlessContext';
 import useLayoutEffect from 'use-isomorphic-layout-effect';
+import { useSyncExternalStoreWithSelector } from 'use-sync-external-store/shim/with-selector';
 
 export type StateSelector<T> = (s: State) => T;
 
+const canUseDOM = !!(typeof window !== 'undefined'
+  && typeof window.document !== 'undefined'
+  && typeof window.document.createElement !== 'undefined');
+const isServerEnvironment = !canUseDOM;
+
+/**
+ * use-sync-external-store/shim does not support getServerSnapshot for pre-18 React versions.
```

Comment:
Could we explain what getServerSnapshot is, or write this comment in a way where people don't have to know?

Reply:
removed `useAnswersStateWithManualSyncStore` so this change is no longer needed